Ok, now it works when `OUTPUT_DIR = .` except for the `rmdir` which still raise an error.

But it does not works when I specify another `OUTPUT_DIR`, 
See the error logs:

```
mkdir -p build
TEXINPUTS="src:" pdflatex -output-directory build \
                                 src/main
This is pdfTeX, Version 3.14159265-2.6-1.40.17 (TeX Live 2016/Arch Linux) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./src/main.tex
LaTeX2e <2016/03/31> patch level 3
Babel <3.9r> and hyphenation patterns for 83 language(s) loaded.
(src/IEEEtran.cls
Document Class: IEEEtran 2012/11/21 V1.8c by Harald Hanche-Olsen and Anders Chr
istensen
-- Based on V1.7a by Michael Shell
-- See the "IEEEtran_HOWTO" manual for usage information.
-- http://www.michaelshell.org/tex/ieeetran/
(/usr/share/texmf-dist/tex/latex/psnfss/ot1ptm.fd)
-- Using 8.5in x 11in (letter) paper.
-- Using PDF output.
-- This is a 10 point document.
) (/usr/share/texmf-dist/tex/latex/blindtext/blindtext.sty
(/usr/share/texmf-dist/tex/latex/tools/xspace.sty))
(/usr/share/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)
(/usr/share/texmf-dist/tex/latex/graphics-def/pdftex.def
(/usr/share/texmf-dist/tex/generic/oberdiek/infwarerr.sty)
(/usr/share/texmf-dist/tex/generic/oberdiek/ltxcmds.sty))))
No file main.aux.
(/usr/share/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texmf-dist/tex/generic/oberdiek/pdftexcmds.sty
(/usr/share/texmf-dist/tex/generic/oberdiek/ifluatex.sty)
(/usr/share/texmf-dist/tex/generic/oberdiek/ifpdf.sty))
(/usr/share/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty
(/usr/share/texmf-dist/tex/latex/oberdiek/grfext.sty
(/usr/share/texmf-dist/tex/generic/oberdiek/kvdefinekeys.sty))
(/usr/share/texmf-dist/tex/latex/oberdiek/kvoptions.sty
(/usr/share/texmf-dist/tex/generic/oberdiek/kvsetkeys.sty
(/usr/share/texmf-dist/tex/generic/oberdiek/etexcmds.sty)))
(/usr/share/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))
(src/chapters/introduction.tex

LaTeX Warning: Citation `theTimelessWayOfBuilding' on page 1 undefined on input
 line 11.


LaTeX Warning: Citation `elementOfReusableObjectOriented' on page 1 undefined o
n input line 11.


LaTeX Warning: Citation `antiPatterns' on page 1 undefined on input line 12.


LaTeX Warning: Citation `improvingDesign' on page 1 undefined on input line 16.



Underfull \hbox (badness 10000) in paragraph at lines 2--21


LaTeX Warning: Citation `aRelationalModelOfDataForLargeSharedDataBanks' on page
 1 undefined on input line 26.


Underfull \hbox (badness 2478) in paragraph at lines 22--38
\OT1/ptm/m/n/10 ta-ble, columns and pri-mary/foreign keys to avoid data

Underfull \hbox (badness 10000) in paragraph at lines 22--38


LaTeX Warning: Reference `problemDescription' on page 1 undefined on input line
 39.


LaTeX Warning: Reference `proposedSolution' on page 1 undefined on input line 4
1.


LaTeX Warning: Reference `discussion' on page 1 undefined on input line 42.


LaTeX Warning: Reference `relatedWork' on page 1 undefined on input line 43.


LaTeX Warning: Reference `conclusion' on page 1 undefined on input line 45.

) (src/chapters/problem_description.tex
Underfull \hbox (badness 10000) in paragraph at lines 4--13

)
No file main.bbl.
[1{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}] [2] (build/main.aux)

LaTeX Warning: There were undefined references.


LaTeX Warning: Label(s) may have changed. Rerun to get cross-references right.

 )
(see the transcript file for additional information){/usr/share/texmf-dist/font
s/enc/dvips/base/8r.enc}</usr/share/texmf-dist/fonts/type1/urw/times/utmb8a.pfb
></usr/share/texmf-dist/fonts/type1/urw/times/utmbi8a.pfb></usr/share/texmf-dis
t/fonts/type1/urw/times/utmr8a.pfb></usr/share/texmf-dist/fonts/type1/urw/times
/utmri8a.pfb>
Output written on build/main.pdf (2 pages, 51407 bytes).
Transcript written on build/main.log.
BIBINPUTS="src:" \
    BSTINPUT="src:" \
    TEXMFOUTPUTS="build:" \
    bibtex main
I couldn't open file name `main.aux'
make: *** [Makefile:38: main] Error 1
```

--- a/Makefile.latex
+++ b/Makefile.latex
@@ -3,45 +3,62 @@
 ##### Sources directory
 SRC_DIR		=
 
-##### file to compile, without tex extension;
+##### Main file, without .tex extension.
 NAME		=
 
 ##### Output directory
 # If not modified, it is the directory which holds the MakeFile
-OUTPUT_DIR = .
+OUTPUT_DIR	=
 
 ##### Executable/output to use: PDFLATEX, DVILUATEX
 CC			= $(PDFLATEX)
+CC_BIB		= $(BIBTEX)
 ################################################################################
 
 ################################################################################
 ############################# Constant variables ###############################
 ##### Shell
-SHELL=/bin/bash
-
-##### Build name file
-SRC			= $(NAME).tex
+SHELL		= /bin/bash
 
 ##### Compilers
 PDFLATEX	= pdflatex
 DVILUATEX	= dviluatex
+BIBTEX		= bibtex
 ################################################################################
 
 ################################################################################
 ############################### Rules ##########################################
-$(NAME):
+all: $(NAME)
+
+$(OUTPUT_DIR):
 	mkdir -p $(OUTPUT_DIR)
-	cd $(SRC_DIR) && $(CC) -output-directory ../$(OUTPUT_DIR) $(SRC)
-	cd $(SRC_DIR) && $(CC) -output-directory ../$(OUTPUT_DIR) $(SRC)
+
+$(NAME): $(OUTPUT_DIR)
+	TEXINPUTS="$(SRC_DIR):" $(CC) -output-directory $(OUTPUT_DIR) \
+                                 $(SRC_DIR)/$(NAME)
+	BIBINPUTS="$(SRC_DIR):" \
+    BSTINPUT="$(SRC_DIR):" \
+    TEXMFOUTPUTS="$(OUTPUT_DIR):" \
+    $(CC_BIB) $(OUTPUT_DIR)/$(NAME)
+	TEXINPUTS="$(SRC_DIR):" $(CC) -output-directory $(OUTPUT_DIR) \
+                                 $(SRC_DIR)/$(NAME)
+	TEXINPUTS="$(SRC_DIR):" $(CC) -output-directory $(OUTPUT_DIR) \
+                                 $(SRC_DIR)/$(NAME)
 
 zip: fclean $(NAME)
 	$(MAKE) clean
 	zip -r $(NAME).zip $(OUTPUT_DIR)/* -x *.git*
 
 clean:
-	$(RM) -f $(OUTPUT_DIR)/$(NAME).{out,aux,toc,log,tex.backup,nav,snm}
+	$(RM) $(OUTPUT_DIR)/$(NAME).{out,aux,toc,log,tex.backup,nav,snm,bbl,blg}
 
+# $(OUTPUT_DIR) is only removed when the directory is empty (never be the case
+# if it's the project root directory).
 fclean: clean
-	$(RM) -f $(OUTPUT_DIR)/$(NAME).{pdf,zip,dvi}
-	rmdir $(OUTPUT_DIR)
+	$(RM) $(OUTPUT_DIR)/$(NAME).{pdf,zip,dvi}
+	-rmdir $(OUTPUT_DIR)
+
+re: fclean $(NAME)
 ################################################################################
+
+.PHONY: all $(NAME) zip clean fclean re

